JustSaying.Models/Message.cs

```diff
@@ -8,11 +8,20 @@ protected Message()
         {
             TimeStamp = DateTime.UtcNow;
             Id = Guid.NewGuid();
+            CorrelationId = Id;
```

Comment:
@MuayThai You are right, and we have this on our roadmap for v6 (unofficially).

See here #99 

I would see it that we add a new envelope wrapper that pulls these framework-y bits off of the base class, and remove the need for the base class altogether. We'd need side-by-side support for at least one version.

For now, ignoring the property clash edge case, it needs to not break existing messages be deserialized to the updated class.